### ISSUE TITLE
[Feature Request] Using Qwen3 VL for semantic guidance in REPA instead of DINO (2 papers came out recently that prove Qwen2.5/3 VL models are much better for video REPA)

### ISSUE THREAD

**Issue by MeiYi-dev** [bghira/SimpleTuner#2247]
https://arxiv.org/pdf/2508.15761
https://arxiv.org/pdf/2512.20619v1

Both of these papers use Qwen2.5 VL (ViT?) for improving semantics in video models immensely.

1st paper uses Qwen2.5 VL for REPA

2nd paper uses Qwen2.5 VL ViT for semantic guidance during inference

I already knew about this before, but now 2 papers prove it working.

**Comment by kabachuha** [comment]
The **very** big negative thing (which is never talked about in these papers!) about REPA is that it requires not only the auxiliary model (DINO/VL), but the **VAE** to stay in GPU and decode the enormous video latents. (And its VRAM usage is clearly visible during the pre-caching phase)

For Kandinsky, I still didn't manage to make it for into one GPU, so I resort to use backbone features (layersync style)

There are things like tiny VAE (TEA) from the community, and they might help, if there were be a flag to select them https://github.com/madebyollin/taehv in the settings, in contrast to the caching VAE selection 👀

The video latents are still huge even for top GPUs, I wonder how complex would it be to implement offloading of VAE to a second GPU in the tuner.

Also, there is backpropagation through these components (though frozen) and it adds HUGE overhead

TAEHV + DINO/VL + 2nd GPU offload combo can probably realise it for big boys like K5

- - -

P.S. CREPA seems to be really powerful, kohya-ss trainer didn't make a good job training the hydraulic press LoRA with the same learning rate, scheduler and even more epochs, while SimpleTuner did it well, see https://github.com/bghira/SimpleTuner/issues/2204#issuecomment-3678872016 for comparison. CREPA similarity was also growing as seen in the plots

**Comment by MeiYi-dev** [comment]
> The **very** big negative thing (which is never talked about in these papers!) about REPA is that it requires not only the auxiliary model (DINO/VL), but the **VAE** to stay in GPU and decode the enormous video latents. (And its VRAM usage is clearly visible during the pre-caching phase)
> 
> For Kandinsky, I still didn't manage to make it for into one GPU, so I resort to use backbone features (layersync style)
> 
> There are things like tiny VAE (TEA) from the community, and they might help, if there were be a flag to select them https://github.com/madebyollin/taehv in the settings, in contrast to the caching VAE selection 👀
> 
> The video latents are still huge even for top GPUs, I wonder how complex would it be to implement offloading of VAE to a second GPU in the tuner.
> 
> Also, there is backpropagation through these components (though frozen) and it adds HUGE overhead
> 
> TAEHV + DINO/VL + 2nd GPU offload combo can probably realise it for big boys like K5
> 
> P.S. CREPA seems to be really powerful, kohya-ss trainer didn't make a good job training the hydraulic press LoRA with the same learning rate, scheduler and even more epochs, while SimpleTuner did it well, see [#2204 (comment)](https://github.com/bghira/SimpleTuner/issues/2204#issuecomment-3678872016) for comparison. CREPA similarity was also growing as seen in the plots

<img width="1641" height="99" alt="Image" src="https://github.com/user-attachments/assets/f2cc4951-6e14-4891-a2fb-ed1d0d130067" />

AFAIK, the VAE latents, TE embeddings and the features extracted from DINO/QWEN ViT can all be pre computed and stored offline. Musubi already does this (except the REPA part) so this isn't a big issue.

**Comment by kabachuha** [comment]
Hmm, looking at the code (https://github.com/bghira/SimpleTuner/blob/main/simpletuner/helpers/training/crepa.py) and on the figures (1) at the REPA paper (https://arxiv.org/pdf/2410.06940), maybe you're right. It compares the hidden states features with decoded source latents which are the input and have not been changed / noised in the generation

SimpleTuner too already caches the video latents and the text encodings, and in the auxiliary loss / CREPA computation it decodes the batch latents with the loaded VAE back and pushes them through DINO. If this process (including DINO) could be precomputed and REPA/CREPA could be used without having VAE and DINO in the VRAM and this would be massive!

Hopefully, I'm not wrong 😬

**Comment by bghira** [comment]
i don't pre-cache the DINO outputs because i didn't want to have to implement it - the embeds are larger than you'd expect for the models' size, and they don't compress down well unlike T5 outputs. currently there exists support for aux image embeds and ODE endpoints but i wouldn't want to reuse either of these because it limits us from eg. applying dino pre-caching when training models that already require aux image embeds, like z-image edit (SigLIP). so, tying the pre-caching specifically to REPA's usage of it would be more work, but cleaner outcome and more maintainable (in my experience).

keeping it on a 2nd GPU and routing requests there would require underlying infrastructure in ST that does not yet exist, and is a pretty big departure from the current paradigm of relying on `accelerate` - not to say it can't be done, it's just not something with a clear end-to-end path yet. probably easier and more "drop-in" if you were to hack it in would be a http client that retrieves the embeds from a remote. then you can operate the aux components from anywhere.

**Comment by kabachuha** [comment]
Thanks for the notes about the DINO embeds!

Though, is it possible to "un-cache" DINO inputs (feed raw videos w/o VAE encode), so the big VAE won't be loaded?

**Comment by bghira** [comment]
is the VAE for k5 still a huge problem even after importing their memory-aware version?

**Comment by kabachuha** [comment]
Yes, it's quite huge either way

TAE-HV may help it, if there were a switch for auxiliary VAE

**Comment by kabachuha** [comment]
And memory-aware sacrifices speed. With the hidden states embeddings compressed (and aligned with noisy states) I think the full VAE version might be an overkill for this alignment

**Comment by kabachuha** [comment]
The inconvenient thing is that the tiny VAEs are not in diffusers format originally https://github.com/madebyollin/taehv/tree/main, consisting of pytorch files + reference launch code (MIT license), so they will require a little bit of tinkering

**Comment by kabachuha** [comment]
Maybe important thought: online VAE decompression of latents introduces its own imperfections and biases, such as granularity

If the REPA model receives processed latents, it will learn the artifacts and they will show in the videos

The only solution for this is, unfortunately, caching the features or loading the video files themselves online

**Comment by bghira** [comment]
there's already tiny autoencoder support in simpletuner for various models, just not plumbed into anything but validation preview streaming

**Comment by kabachuha** [comment]
CREPA's DINO can also benefit from "sliced" processing as we need frame-by-frame features and we can encode the frames in batches instead of the whole video at the same time, sacrificing a bit of speed for lower VRAM

(The current code flattens `batches x frames`)

**Comment by bghira** [comment]
this is implemented in #2285
